x-pack/test/api_integration/apis/ml/filters/get_filters.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { USER } from '../../../../functional/services/ml/security_common';
+import { COMMON_REQUEST_HEADERS } from '../../../../functional/services/ml/common';
+
+// eslint-disable-next-line import/no-default-export
+export default ({ getService }: FtrProviderContext) => {
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+
+  const validFilters = [
+    {
+      filterId: 'filter_1',
+      requestBody: { description: 'Valid filter #1', items: ['104.236.210.185'] },
+    },
+    {
+      filterId: 'filter_2',
+      requestBody: { description: 'Valid filter #2', items: ['104.236.210.185'] },
+    },
+  ];
+
+  describe('get_filters', function () {
+    before(async () => {
+      await ml.testResources.setKibanaTimeZoneToUTC();
+      for (const filter of validFilters) {
+        const { filterId, requestBody } = filter;
+        await ml.api.createFilter(filterId, requestBody);
+      }
+    });
+
+    after(async () => {
+      for (const filter of validFilters) {
+        const { filterId } = filter;
+        await ml.api.deleteFilter(filterId);
+      }
+    });
+    it(`should fetch all filters`, async () => {
+      const { body } = await supertest
+        .get(`/api/ml/filters`)
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(200);
+
+      expect(body).to.have.length(validFilters.length);
+    });
+
+    it(`should not allow to retrieve for user without required permission`, async () => {
+      const { body } = await supertest
+        .get(`/api/ml/filters`)
+        .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(404);
+      expect(body.error).to.eql('Not Found');
+      expect(body.message).to.eql('Not Found');
+    });
+
+    it(`should not allow to retrieve filters for unauthorized user`, async () => {
+      const { body } = await supertest
+        .get(`/api/ml/filters`)
+        .auth(USER.ML_UNAUTHORIZED, ml.securityCommon.getPasswordForUser(USER.ML_UNAUTHORIZED))
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(404);
+
+      expect(body.error).to.eql('Not Found');
+      expect(body.message).to.eql('Not Found');
+    });
+
+    it(`should fetch single analytics job stats by id`, async () => {
```

Comment:
```suggestion
    it(`should fetch single filter by id`, async () => {
```

Reply:
Fixed in https://github.com/elastic/kibana/pull/72564/commits/121d1e5fdd860b91516a4ae3e2023cec45205c9b